src/AggregateFunctions/AggregateFunctionAvg.h

```diff
@@ -42,6 +48,9 @@ struct AvgFraction
     /// Invoked only is either Numerator or Denominator are Decimal.
     Float64 NO_SANITIZE_UNDEFINED divideIfAnyDecimal(UInt32 num_scale, UInt32 denom_scale [[maybe_unused]]) const
     {
+        if (denominator == static_cast<Denominator>(0))
+            throw Exception(ErrorCodes::ILLEGAL_DIVISION, "Cannot calculate average of 0 numbers or all number have 0 weight");
```

Comment:
`avgWeighted` may work with signed weights so it's better to say:
```suggestion
            throw Exception(ErrorCodes::ILLEGAL_DIVISION, "Cannot calculate average value. No numbers provided or sum of all weights is equal to 0");
```